aws_lambda_powertools/metrics/provider/datadog/datadog.py

```diff
@@ -292,4 +299,93 @@ def set_default_tags(self, **tags) -> None:
             def lambda_handler():
                 return True
         """
+        self._validate_datadog_tags_name(**tags)
```

Comment:
```suggestion
        self._validate_datadog_tags_name(tags)
```

performance tip - no need to expand the dict if you control and want a single object. Expanding (splat) will end up creating another object underneath